### ISSUE TITLE
Excessive RAM consumption with large notes

### ISSUE THREAD

**Issue by byohano** [linuxmint/sticky#65]
Hello,

I've started using Sticky Notes recently to organize some of my personal notes (like feedback on my week). Using this software, I noticed my computer's RAM was significantly used when I edited a large note (~100 lines), along with visible lag during input. I suppose it's a bug, but maybe it's expected behaviour? At least, that disturbs my workflow, as I need to stop Sticky Notes (right-click the tray icon --> Click "Quit") and restart it regularly to free up RAM.

### Computer specs
8 GB of RAM according to vendor, actually 7.1GB usable according to my resource monitor
Linux Mint 20.3
Sticky Notes v1.8

### Steps to reproduce
1) Open Sticky Notes, make a new group, create a new note in this group.
2) Write a lot of text. To keep it simple, I just held the A key for a while.
RAM should be very high at this point, and can reach 100% use given enough time.
3) As long as the computer is still responsive, right-click the Sticky notes tray icon, and click "Quit".
The Sticky Notes application should close, and a ton of RAM should be freed.

### What I observed
In less than 5 minutes and around 3800 characters, RAM starts to ramp up dramatically, increasing by a rate close to 60MB each second. Furthermore, while characters appeared normally in the note after pressing keyboard keys, once RAM gets very high, letters appearances start visibly lagging behind physical input.
Attached are some pictures I took of my resource monitor (words are in French language)
![sticky_ram_bug_1](https://user-images.githubusercontent.com/42712324/154091330-27b6084c-094f-4546-b95a-a0d97a659927.png)
This shows RAM usage after I stopped typing in Sticky Notes (past 3800 characters). RAM usage rose to near 100% and the swap memory also increased a bit.
![sticky_ram_bug_2](https://user-images.githubusercontent.com/42712324/154091830-f38c2683-9a0a-4125-bb99-1777e1e1a670.png)
This shows info on the "sticky.py" process, responsible for Sticky Notes (afaik). Memory use in the right-most column is 4.9GB, as confirmed also by virtual memory and resident memory. Shared memory is low, so if I understand it right, the large RAM use is not for communicating with other processes, but only to run the "sticky.py" process.
![sticky_ram_bug_3](https://user-images.githubusercontent.com/42712324/154092636-66ee1b02-dd5d-4bfa-8bc0-630b81e30892.png)
This shows RAM usage after I stopped Sticky Notes normally. Indicated by the blue arrow is the sharp drop in RAM use that followed (80% of max RAM), which seems to be consistent with the disappearance of the "sticky.py" process, which proves Sticky Notes really was allocated this much RAM.

###  What I expected
No significant RAM consumption while editing a note. In comparison, Xed (a text editor) doesn't consume any more RAM after typing over 4000 characters for example. A text editor isn't a note-taking application though, and I don't have any other note-taking app installed other than Sticky Notes (because it ships with Linux Mint versions).

I hope this can be fixed, because I found Sticky Notes to be a nice and intuitive app for note taking.
Let me know if further informations are needed.

**Comment by m8** [comment]
also, memory consumption increases while changing note position. Current ram usage (just reordering my notes):
![image](https://user-images.githubusercontent.com/19881231/155958936-4a68a9f0-9b13-430e-8fc5-0f9c7a21ca64.png)

previous issue about it (can be related): https://github.com/linuxmint/sticky/issues/51#issuecomment-1002775131

**Comment by collinss** [comment]
Quick update: we've tracked down the performance issue, and I have implemented what I believe is a suitable fix. The memory leak is apparently a different issue, and we're still working on that one.

**Comment by m8** [comment]
Hi, I have been testing new version for 2 days, it works fine! Also, I didn't see any major memory leak.

**Comment by byohano** [comment]
I've also updated my version of Sticky Notes with the above fix, and tested a bit, although not rigorously.
I wrote a note of around 6200 characters over 50 minutes.
Previously, Sticky Notes would have consumed at the very least 6 or 7 GB or RAM (actually, I'd most likely have had to restart Sticky Notes before that point),
Here, "sticky.py" used roughly 900MB of virtual memory, 490MB of resident memory, and 440MB of used memory, less than 10% of what would have happened before. From the level of RAM I started with, I noticed a total increase in RAM use of about 6%, which seems to be consistent with my computer's RAM specs (7.1GB of available RAM * 0.06 = 427MB ~= 440MB).
Also didn't see any lag between my typing and a character appearing onscreen, which used to happen when RAM was saturated.
So overall, while there's indeed still a small memory leak, I have no more trouble writing large notes even with other processes in the background, which is already really helpful.
Thank you for the fix! Hope you'll figure out what's causing that memory leak too.
